vpr/src/base/region.h

```diff
@@ -0,0 +1,25 @@
+#ifndef REGION_H
+#define REGION_H
+
+#include "vtr_strong_id.h"
+#include "globals.h"
+
+/**
+ * This class stores the data for each constraint region - the region bounds and the subtile bounds.
+ */
+
+//Type tag for RegionId
+struct region_id_tag;
+
+//A unique identifier for a region
+typedef vtr::StrongId<region_id_tag> RegionId;
+
+class Region {
+  public: //accessors
+  private:
+    //may need to include zmin, zmax for future use in 3D FPGA designs
+    vtr::Rect<int> region_bounds; //xmin, xmax, ymin, ymax
+    int subTilemin, subTilemax;
```

Comment:
Inclusive?  If you don't care about which sub_tile, you can use 0 and Region::HUGE_SUBTILE and all subtiles will be legal.

If you want a specific location, you can specific xmin = xmax = x, ymin=ymax=y, subTilemin = subTilemax = sub_tile.

Reply:
I'd rename subTilemin to sub_tile_min since all other names in this class use underscores.

Reply:
subTileMin is more camelcase.  But other variables use underscores, so I'd just make them all underscore separated for consistency.